src/memphis.ts

```diff
@@ -488,39 +493,115 @@ class Producer {
         }
     }
 
+    private _parseJsonSchema(): any {
+        const ajv = new Ajv();
+        let stationSchemaData = this.connection.stationSchemaDataMap.get(this.internal_station);
+        const schema = stationSchemaData['active_version']['schema_content'];
+        const schemaObj = JSON.parse(schema);
+        let validate: any;
+        try {
+            validate = ajv.compile(schemaObj);
+            return validate;
+        } catch (error) {
+            try {
+                ajv.addMetaSchema(draft7MetaSchema);
+                validate = ajv.compile(schemaObj);
+                return validate;
+            } catch (error) {
+                try {
+                    const ajv = new jsonSchemaDraft04();
+                    validate = ajv.compile(schemaObj);
+                    return validate;
+                } catch (error) {
+                    try {
+                        const ajv = new Ajv2020();
+                        validate = ajv.compile(schemaObj);
+                        return validate;
+                    } catch (error) {
+                        try {
+                            ajv.addMetaSchema(draft6MetaSchema);
+                            return validate;
+                        } catch (error) {
+                            throw MemphisError(new Error('invalid json schema'));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private _validateJsonSchemaMessage(msg: any): any {
+        try {
+            let validate = this._parseJsonSchema();
```

Comment:
You can not parse the schema for each message sent by the user this will kill the performance, it should happen once when a new schema update arrives